Apps/Medication/src/Parsers/TRPParser.cs

```diff
@@ -0,0 +1,130 @@
+﻿//-------------------------------------------------------------------------
+// Copyright © 2019 Province of British Columbia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-------------------------------------------------------------------------
+namespace HealthGateway.MedicationService.Parsers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using HealthGateway.MedicationService.Models;
+    using HL7.Dotnetcore;
+    using Microsoft.Extensions.Configuration;
+
+    /// <summary>
+    /// Parser of TRP (Patient Profile) messages.
+    /// </summary>
+    public static class TrpParser
+    {
+        private const string TRACE = "101010";
+
+        /// <summary>
+        /// Creates a TRP request message to HNClient.
+        /// </summary>
+        /// <param name="phn">The patient phn.</param>
+        /// <param name="userId">The requester user id.</param>
+        /// <param name="ipAddress">The requester ip address phn.</param>
```

Comment:
Is the ip address going to be the Webclient's one?

Reply:
it should be the client requesting the data (the ip from the end user)